shopinvader_cart_reminder/tests/test_sale_order.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from datetime import timedelta
+
+from odoo import api, fields
+from odoo.addons.shopinvader.tests.test_notification import CommonCase
+
+
+class TestSaleOrder(CommonCase):
+    """
+    Tests for sale.order
+    """
+
+    def setUp(self):
+        super(TestSaleOrder, self).setUp()
+        self.sale_obj = self.env["sale.order"]
+        self.sale = self.env.ref("shopinvader.sale_order_2")
+        self.template = self.env.ref(
+            "shopinvader_cart_reminder."
+            "mail_template_shopinvader_sale_reminder"
+        )
+        self.backend.write(
+            {
+                "quotation_reminder": 1,
+                "quotation_reminder_mail_template_id": self.template.id,
+            }
+        )
+
+    def test_reminder1(self):
+        """
+        Test the reminder
+        For this case, the sale should have a reminder
+        :return:
+        """
+        # Ensure not already a reminder
+        self.assertFalse(self.sale.last_reminder)
+
+        @api.model
```

Comment:
@acsonefho There is a lot of duplicate lines...  lines from 40 TO 46 included should be factorized into a method with an optional timedelta as parameter.

Reply:
@lmignon Can you help me? How can I give an optional parameter for a patched method? Thanks :)

Reply:
@acsonefho variable outside the scope of the mehtod when the method is defined are available into the method....

```python
det my_patcher(self):
    my_custom_param = X
    def  patched_method(self):
        return self.do_somthing(my_custom_param)
     MyClass.my_method = patched_method
    ....
```